processing/src/main/java/io/druid/query/groupby/epinephelinae/BufferGrouper.java

```diff
@@ -59,6 +59,16 @@
 public class BufferGrouper<KeyType> implements Grouper<KeyType>
 {
   private static final Logger log = new Logger(BufferGrouper.class);
+  private static final AggregateResult DICTIONARY_FULL = AggregateResult.failure(
+      "Not enough dictionary space to execute this query. Try increasing "
+      + "druid.query.groupBy.maxMergingDictionarySize or enable disk spilling by setting "
+      + "druid.query.groupBy.maxOnDiskStorage to a positive number."
```

Comment:
This error will only be seen by the user if maxOnDiskStorage is 0, so "positive" does make sense. If it's > 0 they'll see a different error, about disk space being full.